controlplane/kubeadm/controllers/controller.go

```diff
@@ -673,3 +684,36 @@ func (r *KubeadmControlPlaneReconciler) adoptOwnedSecrets(ctx context.Context, k
 
 	return nil
 }
+
+// isKubernetesVersionSupported finds out if the Kubernetes version is supported.
+// Kubernetes versions >= v1.22.0 for management clusters are not supported. We identify
+// management clusters by the existence of the KubeadmControlPlane CRD.
+func (r *KubeadmControlPlaneReconciler) isKubernetesVersionSupported(ctx context.Context, cluster *clusterv1.Cluster, version string) (bool, error) {
+	kubernetesVersion, err := util.ParseMajorMinorPatch(version)
+	if err != nil {
+		return false, err
+	}
+
+	// Kubernetes version is < v1.22.0.
+	if kubernetesVersion.LT(semver.MustParse("1.22.0")) {
```

Comment:
Can we only parse the version once in a internal variable?

Reply:
Like `maxKubernetesManagementVersion`

Reply:
Done